pkg/spiffs/fs/spiffs_fs.c

```diff
@@ -139,10 +139,11 @@ static int _mount(vfs_mount_t *mountp)
 #if SPIFFS_SINGLETON == 0
     DEBUG("spiffs: mount: mtd page_size=%" PRIu32 ", pages_per_sector=%" PRIu32
           ", sector_count=%" PRIu32 "\n", dev->page_size, dev->pages_per_sector, dev->sector_count);
-    fs_desc->config.phys_size = dev->page_size * dev->pages_per_sector * dev->sector_count;
+    uint32_t sector_count = (fs_desc->block_count == 0) ? dev->sector_count : fs_desc->block_count;
+    fs_desc->config.phys_size = dev->page_size * dev->pages_per_sector * sector_count;
     fs_desc->config.log_block_size = dev->page_size * dev->pages_per_sector;
     fs_desc->config.log_page_size = dev->page_size;
-    fs_desc->config.phys_addr = 0;
+    fs_desc->config.phys_addr = fs_desc->base_addr;
```

Comment:
please add an assert that checks that the partition does not begin or end outside of the physical device limits.

Reply:
added 2 assert to check:
- `base_addr + sector_count` is inside memory boundaries
- `base_addr` is aligned on a sector